packages/router/src/history.js

```diff
@@ -6,31 +6,26 @@ import { createBrowserHistory } from 'history';
 /**
  * WordPress dependencies
  */
-import { addQueryArgs, getQueryArgs, removeQueryArgs } from '@wordpress/url';
+import { buildQueryString } from '@wordpress/url';
 
 const history = createBrowserHistory();
 
 const originalHistoryPush = history.push;
 const originalHistoryReplace = history.replace;
 
+function buildSearch( params ) {
+	const queryString = buildQueryString( params );
+	return queryString.length > 0 ? '?' + queryString : queryString;
+}
+
 function push( params, state ) {
```

Comment:
`push()` and `replace` are now the same function, the only difference is the original history method. Should we combine them in a single function? The whole thing would be something like:

```
/**
 * External dependencies
 */
import { createBrowserHistory } from 'history';

/**
 * WordPress dependencies
 */
import { buildQueryString } from '@wordpress/url';

const history = createBrowserHistory();

function buildSearch( params ) {
	const queryString = buildQueryString( params );
	return queryString.length > 0 ? '?' + queryString : queryString;
}

function createHistoryFunction( method ) {
	return function ( params, state ) {
		const search = buildSearch( params );
		return method.call( history, { search }, state );
	};
}

history.push = createHistoryFunction( history.push );
history.replace = createHistoryFunction( history.replace );

export default history;
```

Reply:
I don't know, this looks too complex, I prefer the straightforward version.